graphql.go

```diff
@@ -75,6 +76,9 @@ type Schema struct {
 	schema *types.Schema
 	res    *resolvable.Schema
 
+	// TODO: don't necessarily want to force directives.ResolverVisitor?
+	// May be acceptable to implement _one_ of the directive interfaces (e.g. add one for the parse/validate step)?
```

Comment:
Potentially we can solve this problem using different interfaces altogether or implement more methods (e.g. `Type()` etc.). I like your idea about Validation directives that validate before we even try to execute a query.